src/app/clusters/energy-preference-server/energy-preference-server.cpp

```diff
@@ -0,0 +1,224 @@
+    const bool lowPowerSupported = featureMapIsGood && ((ourFeatureMap & to_underlying(Feature::kLowPowerModeSensitivity)) != 0);
+
+    switch (aPath.mAttributeId)
+    {
+    case EnergyBalances::Id:
+        if (!balanceSupported)
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+
+        if (gsDelegate != nullptr)
+        {
+            return aEncoder.EncodeList([endpoint](const auto & encoder) -> CHIP_ERROR {
+                chip::Percent step;
+                char buffer[64];
+                chip::MutableCharSpan label(buffer);
+                size_t index   = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = gsDelegate->GetEnergyBalanceAtIndex(endpoint, index, step, label)) == CHIP_NO_ERROR)
+                {
+                    BalanceStruct::Type balance = { step, Optional<CharSpan>(label) };
+                    ReturnErrorOnFailure(encoder.Encode(balance));
+                    index++;
+                }
+                if (err == CHIP_ERROR_NOT_FOUND)
+                {
+                    return CHIP_NO_ERROR;
+                }
+                return err;
+            });
+        }
+        return CHIP_ERROR_INCORRECT_STATE;
+    case EnergyPriorities::Id:
+        if (balanceSupported == false)
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+
+        if (gsDelegate != nullptr)
+        {
+            return aEncoder.EncodeList([endpoint](const auto & encoder) -> CHIP_ERROR {
+                EnergyPriorityEnum priority;
+                size_t index   = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = gsDelegate->GetEnergyPriorityAtIndex(endpoint, index, priority)) == CHIP_NO_ERROR)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(priority));
+                    index++;
+                }
+                if (err == CHIP_ERROR_NOT_FOUND)
+                {
+                    return CHIP_NO_ERROR;
+                }
+                return err;
+            });
+        }
+        return CHIP_ERROR_INCORRECT_STATE;
+    case LowPowerModeSensitivities::Id:
+        if (lowPowerSupported == false)
+        {
+            return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+        }
+
+        if (gsDelegate != nullptr)
+        {
+            return aEncoder.EncodeList([endpoint](const auto & encoder) -> CHIP_ERROR {
+                chip::Percent step;
+                char buffer[64];
+                chip::MutableCharSpan label(buffer);
+                size_t index   = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = gsDelegate->GetLowPowerModeSensitivityAtIndex(endpoint, index, step, label)) == CHIP_NO_ERROR)
+                {
+                    BalanceStruct::Type balance = { step, Optional<CharSpan>(label) };
```

Comment:
Again, we should have a way for the delegate to indicate "do not include the optional label field".

Reply:
done

Reply:
Now it is really done, with a properly (re-)initialized `Optional<MutableCharSpan>`